### PR TITLE
[SPARK-24263][R] SparkR java check breaks with openjdk

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Change text to grep for.

## How was this patch tested?

manual test

### CODE REVIEW

**Comment by felixcheung** [comment]
@shivaram

**Comment by felixcheung** [comment]
@vanzin FYI - need to fix this before release

File: R/pkg/R/client.R [hunk 1]
```diff
@@ -82,7 +82,7 @@ checkJavaVersion <- function() {
                    })
   javaVersionFilter <- Filter(
       function(x) {
-        grepl("java version", x)
+        grepl(" version", x)
```
Comment (shivaram) [inline R/pkg/R/client.R]: do we want the space " version" vs. "version" that is

**Review by shivaram — APPROVED** [review]
LGTM. Had one minor comment

File: R/pkg/R/client.R (hunk 1 above)
Comment (felixcheung) [reply R/pkg/R/client.R]: right, I debated about it. I think we should grep for the longest match as much as possible but I suppose the space is too common to be meaningful, so I don't feel strongly either way

Comment (shivaram) [reply R/pkg/R/client.R]: Is there any chance you have access to IBM jdk ? I'm just wondering if there is some other java version we are missing

Comment (shivaram) [reply R/pkg/R/client.R]: I managed to get it to work. I think our grep should be fine.
```
root@8c0bc3bb4123:~/ibm-java-x86_64-80# ./jre/bin/java -version
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 8.0.5.15 - pxa6480sr5fp15-20180502_01(SR5 FP15))
IBM J9 VM (build 2.9, JRE 1.8.0 Linux amd64-64 Compressed References 20180425_385365 (JIT enabled, AOT enabled)
OpenJ9   - a7ffbfe
OMR      - a531219
IBM      - 59ef3dc)
JCL - 20180425_01 based on Oracle jdk8u171-b11
```

Comment (felixcheung) [reply R/pkg/R/client.R]: great! interesting it's only broken on openjdk then

**Comment by vanzin** [comment]
Looks like this is good to go?

**Comment by shivaram** [comment]
Yes @felixcheung or @vanzin can you merge this ?

**Comment by vanzin** [comment]
Merging to master / 2.3.

**Pull request closed without merging**